### ISSUE TITLE
Don't disable tool-bar-mode and menu-bar-mode

### ISSUE THREAD

**Issue by skangas** [technomancy/better-defaults#37]
The tool bar and menu are useful for beginners. They help discoverability and make things feel a little bit less foreign and overwhelming when first starting out.

I doubt many experienced users will want to keep them enabled, though there are those who do. The question here is only if it's a good *first* advice to give our new users. In other words, is it good to disable them by default.

They are of course not without issues. For example, they are not very pretty. OTOH, there is some work being done which will hopefully improve that situation (see the recent discussion on making Emacs more pouplar).

Please consider removing the code to disable `tool-bar-mode` and `menu-bar-mode`. Or at least to make that part optional, for example by commenting that code out.

Thanks for considering this, and thanks for the package. (I think some of the defaults you propose are indeed just better and should be upstreamed.)

PS. This was prompted by a recent discussion on emacs-devel starting with: https://lists.gnu.org/archive/html/emacs-devel/2020-04/msg00890.html

**Comment by technomancy** [comment]
I disagree about the menu bar; the familiarity it gives users early on is a false sense of familiarity. Users are used to using programs where all the functionality they need is available in the menu. This is absolutely not the case in Emacs.

When the menu bar is on, people will use it to look for commands, and when they don't find the commands they want, they will assume they don't exist. When the menu bar is off, they will learn about commands like `apropos`, or the fact that the `M-x` prompt has tab completion, and thus they will actually be able to find what they are looking for.

Of course, anyone who understands the problems and wants the menu bar anyway can turn on menu-bar-mode, so it is still optional.

**Comment by cpitclaudel** [comment]
> This is absolutely not the case in Emacs.

That's fair, but arguably it's a bug, which we should fix.  (Also, I think this applies more to built-in menus than to 3rd-party packages: many of them make efforts to have useful menus.

(I wonder to what extent the phenomenon feeds on itself: everyone disables the menus, so no one sees them and so there's no incentive to make them better)

Thanks for your work.

**Comment by skangas** [comment]
> I disagree about the menu bar; the familiarity it gives users early on is a false sense of familiarity. Users are used to using programs where all the functionality they need is available in the menu. This is absolutely not the case in Emacs.

Sure, after a while you will have to learn to not rely (too much) on these things. But these settings are making it harder for people to get to the point where they can even learn to use the help commands.

> When the menu bar is on, people will use it to look for commands, and when they don't find the commands they want, they will assume they don't exist. When the menu bar is off, they will learn about commands like `apropos`, or the fact that the `M-x` prompt has tab completion, and thus they will actually be able to find what they are looking for.

How will they learn? It's not an automatic process that will ensue just because they turn off the menus. They will (or should) read about it in the introductory material, the tutorial and the manual. There are two ways to find that help besides knowing the commands: the splash screen and the menu.

Having the menu and tool bar enabled makes it easier, not harder to learn the help commands you are talking about. If you have proposals for how they could do a better job at that, I urge you to send bug reports.

**Comment by filiden** [comment]
@skangas In Emacs, the user interacts with the interface almost only with the keyboard. On the visual aspect, one can obtain a rendering similar to a classic graphical interface but the user interaction is not done with graphical objects. These are two forms of event-driven programming. In practice, the menu and toolbar in Emacs are unsuitable although they seem quite usual at first glance but it's not operational (no graphical windows). Can we really use the mouse to interact with Emacs? I don't think so.
